### PR TITLE
Legends for color scales should use gradient.

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

**Comment by kanitw** [comment]
LGTM.  Any chance you get to revise  https://github.com/vega/vega-lite/issues/1772 too?
(Probably worth looking at now with less context switch in the future? -- I'm fine if you wanna ignore this comment though.)

**Comment by domoritz** [comment]
Done.

**Comment by domoritz** [comment]
Resolved conflicts.

File: src/compile/legend/rules.ts [hunk 1]
```diff
@@ -30,8 +31,7 @@ export function type(legend: Legend, fieldDef: FieldDef, channel: Channel) {
   if (legend.type) {
     return legend.type;
   }
-
-  if (channel === COLOR && !fieldDef.bin && !fieldDef.timeUnit && contains([QUANTITATIVE, TEMPORAL], fieldDef.type)) {
+  if (channel === COLOR && !fieldDef.bin && (!fieldDef.timeUnit || containsTimeUnit(fieldDef.timeUnit, 'year')) && contains([QUANTITATIVE, TEMPORAL], fieldDef.type)) {
```
Comment (kanitw) [inline src/compile/legend/rules.ts]: Should we use scale type here (scale.type === `time` or `utc`)?  (Instead of checking for time prefix directly)

**Comment by domoritz** [comment]
Done.

File: src/compile/legend/rules.ts (hunk 1 above)
Comment (domoritz) [inline src/compile/legend/rules.ts]: Done and with more tests as well.

File: src/compile/legend/rules.ts [hunk 2]
```diff
@@ -26,12 +27,11 @@ export function values(legend: Legend) {
...
 }
 
-export function type(legend: Legend, fieldDef: FieldDef, channel: Channel) {
+export function type(legend: Legend, timeUnit: TimeUnit, channel: Channel, scaleType: ScaleType) {
   if (legend.type) {
     return legend.type;
   }
-
-  if (channel === COLOR && !fieldDef.bin && !fieldDef.timeUnit && contains([QUANTITATIVE, TEMPORAL], fieldDef.type)) {
+  if (channel === COLOR && (!timeUnit || containsTimeUnit(timeUnit, 'year')) && contains<ScaleType>(['time', 'utc', 'sequential'], scaleType)) {
```
Comment (kanitw) [inline src/compile/legend/rules.ts]: I meant replace  `containsTimeUnit(timeUnit, 'year')` with scale type check just for temporal case.  

I think the new logic seems wrong.   You can sensibly use linear, log, pow with quantitative field too.  I think the old `contains([QUANTITATIVE, TEMPORAL], fieldDef.type)` was sensible.

Comment (domoritz) [inline src/compile/legend/rules.ts]: Ahh, I see.

**Pull request merged**